modules/geo-layers/src/mvt-layer/mvt-layer.js

```diff
@@ -267,5 +279,27 @@ function transformTileCoordsToWGS84(object, tile, viewport) {
   return feature;
 }
 
+function conversionToWGS84isRequired(tileCoords, isWGS84) {
+  return tileCoords === 'wgs84' && !isWGS84;
+}
+
+function updateTilesWithWGS84Coords(tiles, tilesData, viewport) {
+  tiles.forEach((tile, idx) => {
+    tile.dataWithWGS84Coords = tilesData[idx].map(object =>
```

Comment:
- The tile instance should never be mutated, let alone conditionally. Bear in mind that MVTLayer is not the only user of TileLayer, while the WGS84 issue is MVT specific. This will also introduce a lot of issues when we move to TypeScript. You can add a method `get dataInWorldCoordinates()` to the `Tile2DHeader` class.
- This seems to be recalculated every time you call `getVisibleTiles`

Reply:
Didn't know about the migration to TS :+1: